tests/test_is_satisfying_hint.py

```diff
@@ -1,8 +1,16 @@
 from typing import Any, Generic, Literal, TypeAlias, TypeVar
 
+from xdsl.dialects.arith import (
```

Comment:
Can you please replace the arith use with a custom operation defined for this unit test? We try to avoid using non-builtin dialects for unit tests

Reply:
That sounds like a clean way of testing. I replaced usages of the arith package with custom ops, and moved the code testing arith ops to the arith test file.